timemachine/fe/chiral_utils.py

```diff
@@ -132,6 +142,131 @@ def find_chiral_atoms(mol):
+        for ii in idxs:
+            assert ii not in chiral_atom_restr_idxs
+        chiral_atom_restr_idxs.extend(idxs)
+    return chiral_atom_restr_idxs
+
+
+@dataclass
+class ChiralRestrIdxSet:
+    """Support fast checks of whether a given 4-tuple is disallowed"""
+
+    restr_idxs: List[FourTuple]
+
+    @classmethod
+    def from_mol(cls, mol, conf):
+        restr_idxs = setup_all_chiral_atom_restr_idxs(mol, conf)
+        return ChiralRestrIdxSet(restr_idxs)
+
+    def expand_symmetries(self) -> Tuple[Set[FourTuple], Set[FourTuple]]:
+        allowed_set = set()
+        disallowed_set = set()
+
+        for (center, i, j, k) in self.restr_idxs:
+            # rotations
+            allowed_set.add((center, i, j, k))
+            allowed_set.add((center, j, k, i))
+            allowed_set.add((center, k, i, j))
+
+            # swaps
+            disallowed_set.add((center, i, k, j))
+            disallowed_set.add((center, j, i, k))
+            disallowed_set.add((center, k, j, i))
+
+        assert allowed_set.isdisjoint(disallowed_set)
+
+        return allowed_set, disallowed_set
+
+    def __post_init__(self):
+        self.allowed_set, self.disallowed_set = self.expand_symmetries()
+
+    def defines(self, trial_tuple: FourTuple) -> bool:
+        return (trial_tuple in self.allowed_set) or (trial_tuple in self.disallowed_set)
+
+    def disallows(self, trial_tuple: FourTuple) -> bool:
+        return trial_tuple in self.disallowed_set
+
+
+def _find_atom_map_chiral_conflicts_one_direction(
+    core: np.ndarray,
+    chiral_set_a: ChiralRestrIdxSet,
+    chiral_set_b: ChiralRestrIdxSet,
+    mode="flip",
+):
+    # parse mode
+    if mode == "flip":
+        conflict_condition_fxn = chiral_set_b.disallows
+    elif mode == "undefined":
+        conflict_condition_fxn = lambda mapped_tuple_b: not chiral_set_b.defines(mapped_tuple_b)
+    else:
+        raise ValueError("invalid mode -- must be one of 'flip' or 'undefined")
+
+    # initialize convenient representations
+    mapped_set_a = set(core[:, 0])
+    conflicts = set()
+    restr_tuples_a = [(int(c), int(i), int(j), int(k)) for (c, i, j, k) in chiral_set_a.restr_idxs]
+    mapping_a_to_b = {int(a_i): int(b_i) for (a_i, b_i) in core}
+
+    def apply_mapping(c, i, j, k):
+        return mapping_a_to_b[c], mapping_a_to_b[i], mapping_a_to_b[j], mapping_a_to_b[k]
+
+    # iterate over restraints defined in A, searching for possible conflicts
+    for restr_tuple_a in restr_tuples_a:
+        if set(restr_tuple_a).issubset(mapped_set_a):
+            mapped_tuple_b = apply_mapping(*restr_tuple_a)
+
+            if conflict_condition_fxn(mapped_tuple_b):
+                conflicts.add((restr_tuple_a, mapped_tuple_b))
+
+    return conflicts
+
+
+def find_atom_map_chiral_conflicts(
+    core: np.ndarray,
+    chiral_set_a: ChiralRestrIdxSet,
+    chiral_set_b: ChiralRestrIdxSet,
+    mode="flip",
+) -> Set[Tuple[FourTuple, FourTuple]]:
+    """
+
```

Comment:
Add a note that chiral here treats all atoms as distinguishable (methane is chiral).

Reply:
Added references to definition + more context for why we adopted this definition in https://github.com/proteneer/timemachine/pull/908/commits/71f83c012c6da720c18503adaa8bf5610977295f